summingbird-batch/src/main/scala/com/twitter/summingbird/batch/BatchID.scala

```diff
@@ -57,8 +69,18 @@ object BatchID {
    * `[startBatch, endBatch]` (inclusive).
    */
   def range(start: BatchID, end: BatchID): Iterable[BatchID] =
-    new Iterable[BatchID] {
-      def iterator = iterate(start)(_.next).takeWhile(_ <= end)
+    if (start == end) List(start)
+    else if (end < start) Nil
+    else {
+      new Iterable[BatchID] {
```

Comment:
Can we keep old way to iterate, maybe with additional check on the `end == Long.MAX_VALUE`? I think it makes code simpler.